### ISSUE TITLE
[bug] Module @storybook/test-runner/playwright/transform in the transform option was not found

### ISSUE THREAD

**Issue by ddubrava** [storybookjs/test-runner#356]
**Describe the bug**

In `"@storybook/test-runner": "0.13.0"`, the script `test-storybook --junit --ci` in NPM workspaces fails with the following error.

```
● Validation Error:

Module @storybook/test-runner/playwright/transform in the transform option was not found.
     <rootDir> is: /Users/ddubrava/Desktop/test-runner-repro
```

It works in `"@storybook/test-runner": "0.12.0"`, so I think the problem might be due to these changes

https://github.com/storybookjs/test-runner/blob/e1455258f3b0f6cb7845f8ca9ed4dad2e9ad6f18/src/config/jest-playwright.ts#L4

**To Reproduce**

I created the reproduction

https://github.com/ddubrava/storybook-test-runner-reproduction

**Expected behavior**

No errors.

**Screenshots**

<img width="723" alt="Screenshot 2023-09-15 at 22 44 35" src="https://github.com/storybookjs/test-runner/assets/22116465/4cab8834-0e3f-47af-af7a-ddc3cadcae8a">

**System**

```
  System:
    OS: macOS 13.4
    CPU: (10) arm64 Apple M1 Pro
  Binaries:
    Node: 18.17.1 - ~/.nvm/versions/node/v18.17.1/bin/node
    npm: 9.6.7 - ~/.nvm/versions/node/v18.17.1/bin/npm
  Browsers:
    Safari: 16.5
```

**Additional context**

I don't know why but sometimes this error doesn't arise even in my repro.

**Comment by ddubrava** [comment]
The temporary fix is to use the version <= `12.0.0`

**Comment by flochtililoch** [comment]
I'm also reproducing this error while running `0.9.4`.
